File: Aman/Arrays & strings/Maximum Number of Vowels in a Substring of Given Length.cpp
class Solution {
public:
    bool isvowel(char c)
    {
        return (c=='a' || c=='e' || c=='i' || c=='o' || c=='u');
    }
    
    int maxVowels(string s, int k) {
        
        int n=s.length();
        int i=0,j=0,cnt=0,res=0;
        
        while(j<k)
        {
             if(isvowel(s[j]))
                cnt++;
             j++;
        }
        res=cnt;
        while(j<n)
        {
            if(isvowel(s[i]))
                cnt--;
            if(isvowel(s[j]))
                cnt++;
            res=max(cnt,res);
            i++;
            j++;
        }
        return res;
    }
};

File: Harish Kumar/Array and Strings/ Maximum Number of Vowels in a Substring of Given Length/ Maximum Number of Vowels in a Substring of Given Length.java
class Solution {
    private boolean isVowel(char c)
    {
        if(c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' )
        {
            return true;
        }
        return false;
    }
    public int maxVowels(String s, int k) {
        
        int length = 0;
        int count  = 0;
        int i = 0;
        
        while( i < k)
        {
            if(isVowel(s.charAt(i)))
            {
                count++;
            }
            i++;
        }
        
        for(; i < s.length();i++)
        {
            if(count > length)
            {
                length = count;
            }
            if(isVowel(s.charAt(i)))
            {
                count++;
            }
            if(isVowel(s.charAt(i -k)))
            {
                count--;
            }
        }
         if(count > length)
            {
                length = count;
            }
        return length;
    }
}
